File: Wire-ScriptCore/Source/Wire/Input.cs
namespace Wire
{
	public class Input
	{
		public static bool IsKeyDown(KeyCode keycode)
		{
			return InternalCalls.Input_IsKeyDown(keycode);
		}

		public static bool IsMouseButtonPressed(MouseButton button)
		{
			return InternalCalls.Input_IsMouseButtonPressed(button);
		}

		public static (float X, float Y) GetMousePosition()
		{
			return (InternalCalls.Input_GetMouseX(), InternalCalls.Input_GetMouseY());
		}

		public static float GetMouseX()
		{
			return InternalCalls.Input_GetMouseX();
		}

		public static float GetMouseY()
		{
			return InternalCalls.Input_GetMouseY();
		}
	}
}

File: Wire/src/Wire/Core/Input.cpp
#include "Input.h"

#include <glfw/glfw3.h>

namespace wire {

	bool Input::isKeyDown(KeyCode key)
	{
		return glfwGetKey(s_Window, static_cast<int>(key)) == GLFW_PRESS;
	}

	bool Input::isMouseButtonDown(MouseButton mouseButton)
	{
		return glfwGetMouseButton(s_Window, static_cast<int>(mouseButton)) == GLFW_PRESS;
	}

	glm::vec2 Input::getMousePosition()
	{
		double x, y;
		glfwGetCursorPos(s_Window, &x, &y);
		return { (float)x, (float)y };
	}

}
